### DISCUSSION TITLE
Does this repository contain the source code of generators (custom tools)?

### DISCUSSION THREAD

**Discussion by Takym** [dotnet/runtime#55078] [Q&A] [answered]
Hello

This is relating to #54117. Is there the source code of generators (custom tools) such as `ResXFileCodeGenerator` or `TextTemplatingFileGenerator`? If the runtime or SDK does not contain generators, how could the files be converted in Linux when I specified `<Generator>...</Generator>` into the `Item` element of the project file?

Thanks

PS: I created a simple small C# project with the `.resx` file. Then, I built the project with the command `dotnet build`. The `dotnet` command did not generate `designer.cs` file. So, is the `<Generator>...</Generator>` only able in Visual Studio?

**Comment by huoyaoyuan** [comment]
Some of the dotnet repo specific tools live in dotnet/arcade repo. You may search there.

**Reply by Takym** [reply]
Thank you for telling me, but I could not find it.
https://github.com/dotnet/arcade/search?q=ResXFileCodeGenerator
https://github.com/dotnet/arcade/search?q=TextTemplatingFileGenerator

**Comment by huoyaoyuan** [comment, marked as answer]
Use MSBuild to find what it uses should be a good approach.

Right click VS project and select "Open in Terminal", the opened terminal will use MSBuild shipped with Visual Studio.

```
msbuild /pp:pp.xml
```
This will process the project file and import everything required. You can search for the xml to find what are imported.

```
msbuild /bl:bl.binlog
```
This will do a build, then log every step MSBuild does. Then you can use the [Log Viewer](https://msbuildlog.com/) to inspect it. The log should contain some details of everything invoked.

**Reply by Takym** [reply]
I looked at `pp.xml` and `bl.binlog`. So, I could confirm that the build system uses `ResGen.exe` for embedded resources and does not generate files with generators (custom tools). Also, I noticed resource designer files or text template outputs are generated **only with Visual Studio on saving them**. Thanks!
